### PR TITLE
(feat) O3-3054 - ward - render empty beds

### PR DESCRIPTION
**Pull request by chibongho**
## Requirements

- [x] This PR has a title that briefly describes the work done including the ticket number. If there is a ticket, make sure your PR title includes a [conventional commit](https://o3-docs.openmrs.org/docs/frontend-modules/contributing.en-US#contributing-guidelines) label. See existing PR titles for inspiration.
- [x] My work conforms to the [OpenMRS 3.0 Styleguide](https://om.rs/styleguide) and [design documentation](https://om.rs/o3ui).
- [x] My work includes tests or is validated by existing tests.

## Summary

This PR adds:
- hooks to get beds and patient bed assignments from the bedmagement server module 
- renders empty beds in a location

## Screenshots

image
image

## Other

Thanks @kb019 for the collaboration!

This PR does not check whether the bedmagement module is installed. Further work to make it an optional dependency is tracked in [O3-3301](https://openmrs.atlassian.net/browse/O3-3301)

[O3-3301]: https://openmrs.atlassian.net/browse/O3-3301?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by kb019** [comment]
Thank you for the guidence @chibongho.

File: packages/esm-ward-app/src/ward-view/ward-view.component.tsx [hunk 1]
```diff
@@ -11,22 +17,76 @@ const WardView = () => {
...
+              kind="warning"
+              lowContrast={true}
+              title={t('noBedsConfigured', 'No beds configured for this location')}
+            />
+          )}
+        </div>
+      </div>
+    );
+  } else if (isLoading) {
+    // TODO: add WardBedSkeleton
```
Comment (mseaton) [inline packages/esm-ward-app/src/ward-view/ward-view.component.tsx:71]: Why is this a todo - is it pending a decision?  Can we implement and get rid of the hard-coded english text?

**Review by mseaton — COMMENTED** [review]
Generally looks good, a few relatively minor comments

File: packages/esm-ward-app/src/ward-view/ward-view.component.tsx [hunk 2]
```diff
@@ -11,22 +17,76 @@ const WardView = () => {
...
-    </div>
-  );
+const WardViewByLocation = ({ location }: { location: Location }) => {
+  const { admissionLocation, isLoading, error } = useAdmissionLocation(location.uuid);
+  const { t } = useTranslation();
+
+  if (admissionLocation) {
+    // admissionLocation.bedLayouts can contain row+column positions with no bed,
+    // filter out layout positions with no real bed
+    const bedLayouts = admissionLocation.bedLayouts.filter((bl) => bl.bedId);
```
[the remark this answers is not in the document]
Comment (mseaton) [reply packages/esm-ward-app/src/ward-view/ward-view.component.tsx:47]: Interesting.  There is definitely no guaranteed these will be strictly numeric.

**Comment by ojwanganto** [comment]
Good stuff @chibongho!

**Review by mseaton — APPROVED** [review]

**Pull request merged**